src/libraries/Microsoft.Extensions.Configuration.Xml/src/XmlConfigurationElement.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.Xml
+{
+    internal class XmlConfigurationElement
+    {
+        public string ElementName { get; }
+
+        public string Name { get; }
+
+        public string LineInfo { get; }
+
+        /// <summary>
+        /// The children of this element
+        /// </summary>
+        public List<XmlConfigurationElement>? Children { get; set; }
+
+        /// <summary>
+        /// The siblings of this element, including itself
+        /// Elements are considered siblings if they share the same element name and name attribute
+        /// This list is shared by each sibling
+        /// </summary>
+        public List<XmlConfigurationElement>? Siblings { get; set; }
+
+        public XmlConfigurationElementTextContent? TextContent { get; set; }
+
+        public List<XmlConfigurationElementAttributeValue>? Attributes { get; set; }
+
+        public XmlConfigurationElement(string elementName, string name, string lineInfo)
+        {
+            ElementName = elementName ?? throw new ArgumentNullException(nameof(elementName));
+            Name = name;
+            LineInfo = lineInfo;
+            Children = null;
```

Comment:
all these would already be null, right?

Reply:
There is no need indeed, but I generally like to be explicit about it. I'll remove those lines since they're not the norm in this repo.